Examples/Algorithms/Geant4/src/ParticleTrackingAction.cpp

```diff
@@ -83,3 +96,34 @@ ActsExamples::SimParticle ActsExamples::ParticleTrackingAction::convert(
   aParticle.setAbsoluteMomentum(p);
   return aParticle;
 }
+
+std::optional<ActsExamples::SimBarcode>
+ActsExamples::ParticleTrackingAction::makeParticleId(G4int trackId,
+                                                     G4int parentId) const {
+  auto& ed = EventStoreRegistry::eventData();
+
+  // We already have this particle registered (it is one of the input particles
+  // or we are making a final particle state)
+  if (ed.trackIdMapping.find(trackId) != ed.trackIdMapping.end()) {
+    return ed.trackIdMapping.at(trackId);
+  }
+
+  if (ed.trackIdMapping.find(parentId) == ed.trackIdMapping.end()) {
+    ACTS_WARNING("Parent particle "
+                 << parentId
+                 << " not registered, maybe because of previous particle "
+                    "collision?");
+    return std::nullopt;
+  }
+
+  auto pid = ed.trackIdMapping.at(parentId).makeDescendant();
+
+  auto key = EventStoreRegistry::State::BarcodeWithoutSubparticle::Zeros();
+  key.set(0, pid.vertexPrimary())
+      .set(1, pid.vertexSecondary())
+      .set(2, pid.particle())
+      .set(3, pid.generation());
```

Comment:
can you explain what this key is?

Reply:
The idea is that we maintain a map, that does hold the current subparticle count for a spezific combination of primary, secondary, particle and generation. This is most easily done by using another `Acts::MultiIndex` object with only four fields as key.